File: Project_ReactNative/src/main/java/com/example/project_reactnative/model/entity/Booking.java
package com.example.project_reactnative.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Booking {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDate startDate;
    private LocalDate endDate;

    private int adults;
    private int children;
    private int infants;

    private String paymentMethod;

    @Enumerated(EnumType.STRING)
    private Status status = Status.INCOMING;
}

File: Project/interface/booking.ts
export type Status = "INCOMING" | "PAST" | "CANCELLED";
export interface Booking {
  id: number;
  startDate: string; 
  endDate: string; 
  adults: number;
  children: number;
  infants: number;
  paymentMethod?: string; 
  status: Status; 
}